server.py

```diff
@@ -28,13 +30,24 @@ def updateBoard(self,imagedata):
 }
 server = Server(imagedata)
 
-@socketio.on('connect')
-def on_connect():
-    print("connected to websocket")
+@socketio.on('connect', namespace='/canvas')
+def connect_canvas():
+    # broadcast board upon initial connect at /canvas endpoint
+    # turn board into JSON
+    board = CanvasBoardEncoder().encode(server.board)
+    emit('broadcast-board', board)
 
 @socketio.on('disconnect')
 def on_disconnect():
     print("disconnected from websocket")
 
+@socketio.on('send-stroke', namespace="/canvas")
+def handle_send_stroke(diff):
+    # diff -> dict
+    server.updateBoard(diff)
+    # turn board into JSON
+    board = CanvasBoardEncoder().encode(server.board)
+    emit('broadcast-board', board)

```

Comment:
Does this actually broadcast to all clients or just the one who sent the message?  (it should be all clients)

Reply:
As of right now, it only broadcasts to the client sending the message. I believe 'rooms' need to be implemented for all the clients to receive the board